src/main/java/gregtech/api/metatileentity/SimpleMachineMetaTileEntity.java

```diff
@@ -90,12 +91,30 @@ public boolean onWrenchClick(EntityPlayer playerIn, EnumHand hand, EnumFacing fa
             }
             if(!getWorld().isRemote) {
                 setOutputFacing(facing);
+                if (!isAllowInputFromOutputSide()) {
+                    if (getCoverCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, facing) != null ||
+                        (getCoverCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, facing) != null)) {
+                        setAllowInputFromOutputSide(true);
+                    }
+                }
             }
             return true;
         }
         return super.onWrenchClick(playerIn, hand, facing, hitResult);
     }
 
+    @Override
+    public boolean placeCoverOnSide(EnumFacing side, ItemStack itemStack, CoverDefinition coverDefinition) {
+        boolean coverPlaced = super.placeCoverOnSide(side,itemStack,coverDefinition);
+        if (getOutputFacing() == side && !isAllowInputFromOutputSide()) {
+            if (getCoverCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY,side) != null ||
+                (getCoverCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY,side) != null)) {
+                setAllowInputFromOutputSide(true);
```

Comment:
Calling `getCoverCapability` here is overkill as it does quite lot of stuff underneath which in this case is not needed or even wanted. It would be simpler just to `getCoverAtSide` and check if for existence of Interface or even call function on `CoverBehavior` which will return info if this cover should `setAllowInputFromOutputSide`.